domain/src/main/java/com/teamagam/gimelgimel/domain/dynamicLayers/DisplayDynamicLayersInteractor.java

```diff
@@ -7,28 +7,42 @@
 import com.teamagam.gimelgimel.domain.base.interactors.BaseSingleDisplayInteractor;
 import com.teamagam.gimelgimel.domain.base.interactors.DisplaySubscriptionRequest;
 import com.teamagam.gimelgimel.domain.dynamicLayers.entity.DynamicLayer;
+import com.teamagam.gimelgimel.domain.dynamicLayers.repository.DynamicLayerVisibilityRepository;
 import com.teamagam.gimelgimel.domain.dynamicLayers.repository.DynamicLayersRepository;
 
 @AutoFactory
 public class DisplayDynamicLayersInteractor extends BaseSingleDisplayInteractor {
-  private DynamicLayersRepository mDynamicLayersRepository;
-  private Displayer mDisplayer;
+  private final DynamicLayersRepository mDynamicLayersRepository;
+  private final DynamicLayerVisibilityRepository mDynamicLayerVisibilityRepository;
+  private final Displayer mDisplayer;
 
   public DisplayDynamicLayersInteractor(@Provided ThreadExecutor threadExecutor,
       @Provided PostExecutionThread postExecutionThread,
       @Provided DynamicLayersRepository dynamicLayersRepository,
+      @Provided DynamicLayerVisibilityRepository dynamicLayerVisibilityRepository,
       Displayer displayer) {
     super(threadExecutor, postExecutionThread);
     mDynamicLayersRepository = dynamicLayersRepository;
+    mDynamicLayerVisibilityRepository = dynamicLayerVisibilityRepository;
     mDisplayer = displayer;
   }
 
   @Override
   protected SubscriptionRequest buildSubscriptionRequest(DisplaySubscriptionRequest.DisplaySubscriptionRequestFactory factory) {
-    return factory.createSimple(mDynamicLayersRepository.getObservable(), mDisplayer::display);
+    return factory.createSimple(
+        mDynamicLayerVisibilityRepository.getChangesObservable().map(this::toPresentation),
```

Comment:
I know it's not necessary performance-wise, but I think I expect this to look like `factory.create(mDynamicLayerVisibilityRepository.getChangesObservable(), o -> o.map(...), mDisplayer::display)`